src/unwinder/i386/unwinder_i386.cpp

```diff
@@ -105,8 +105,19 @@ OOPStackUnwinderX86::VirtualUnwind(
     ENUM_CALLEE_SAVED_REGISTERS();
 #undef CALLEE_SAVED_REGISTER
 
-    ContextRecord->Esp = rd.SP - codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
-    ContextRecord->ResumeEsp = rd.SP;
+    SIZE_T paramSize = codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
+    SIZE_T paddingSize = 0;
+
+#ifdef UNIX_X86_ABI
+    // On UNIX_X86_ABI, function call may have stack alignment padding.
+    if (paramSize % 16 != 0)
```

Comment:
I think we can assume that stack is aligned by local variables with https://github.com/dotnet/coreclr/blob/master/src/jit/lclvars.cpp#L5957 code block.
So for JITted code we won't get any code like
```
push ebp
mov ebp, esp
push ebx
// Now the esp value ends with 4 (cause of return address in stack)
mov eax, arg
push eax
call fn
```
but actually we will get
```
push ebp
mov ebp, esp
sub esp, 8
// Now the esp value ends with 0
sub esp, 0x0c
mov eax, arg
push eax
call fn
```
If there are any saving registers like `push ebx` we have have related codes.